pyboy/plugins/game_wrapper_pokemon_pinball.py

```diff
@@ -0,0 +1,1504 @@
+        """
+
+        if stage is None:
+            return
+        for i in range(NO_OP_BYTE_WIDTH_STAGE_OVERRIDE):
+            # equivalent to no op
+            self.pyboy.memory[ADDR_TO_NO_OP_BANK_STAGE_OVERRIDE, ADDR_TO_NO_OP_STAGE_OVERRIDE + i] = 0x00
+        # equivalent to ld a, stage.value
+        self.pyboy.memory[ADDR_TO_NO_OP_BANK_STAGE_OVERRIDE, ADDR_TO_NO_OP_STAGE_OVERRIDE] = 0b00111110
+        self.pyboy.memory[ADDR_TO_NO_OP_BANK_STAGE_OVERRIDE, ADDR_TO_NO_OP_STAGE_OVERRIDE + 1] = stage.value
+
+    def _init_bonus_stage(self, stage):
+        # set backup stage if it is a bonus stage
+        if stage in RedBonusStages:
+            self.pyboy.memory[ADDR_CURRENT_STAGE_BACKUP] = Stage.RED_BOTTOM.value
+        elif stage in BlueBonusStages:
+            self.pyboy.memory[ADDR_CURRENT_STAGE_BACKUP] = Stage.BLUE_BOTTOM.value
+
+        # do initializations skipped by loading bonus stage instead of main stage
+        if stage in RedBonusStages or stage in BlueBonusStages:
+            self.pyboy.memory[ADDR_CURRENT_SLOT_FRAME] = CURRENT_SLOT_FRAME_VALUE
+            self.pyboy.memory[ADDR_BALLS_LEFT] = 1
+            self.pyboy.memory[ADDR_NUM_BALL_LIVES] = 3
+            self.pyboy.memory[ADDR_STAGE_COLLISION_STATE] = 0b100
+            self.pyboy.memory[ADDR_STAGE_COLLISION_STATE_HELPER] = 0b100
+
+    def start_catch_mode(self, pokemon=Pokemon.BULBASAUR, unlimited_time=False):
+        """
+        Starts the catch mode in the game. NOTE: This method does not change stage specific values and may need a top/bottom stage change to work properly.
+
+        This function sets up the game state for catch mode, including the Pokemon to catch and the game timer.
+
+        Parameters:
+        pokemon (Pokemon, optional): The Pokemon to catch in this mode. Defaults to Pokemon.BULBASAUR.
+        unlimited_time (bool, optional): If True, the game timer is not activated, giving unlimited time in catch mode. Defaults to False.
+
+        Returns:
+        None
+        """
+        # All values are based on PRET disassembly
+        self.pyboy.memory[ADDR_SPECIAL_MODE] = SpecialMode.CATCH.value
+        self.pyboy.memory[ADDR_POKEMON_TO_CATCH] = pokemon.value
+        self.pyboy.memory[ADDR_SPECIAL_MODE_ACTIVE] = 1
+        self.pyboy.memory[ADDR_SPECIAL_MODE_STATE] = 0
+        self.pyboy.memory[0xd5c6] = 0
+        self.pyboy.memory[ADDR_NUM_MON_HITS] = 0
+        self.pyboy.memory[ADDR_NUM_CATCH_TILES_FLIPPED] = 0
+        for i in range(TILE_ILLUMINATION_BYTE_WIDTH):
+            self.pyboy.memory[ADDR_TILE_ILLUMINATION + i] = 0
+        if not unlimited_time:
+            self.pyboy.memory[ADDR_TIMER_SECONDS] = 0
+            self.pyboy.memory[ADDR_TIMER_MINUTES] = 2
+            self.pyboy.memory[ADDR_TIMER_FRAMES] = 0
+            self.pyboy.memory[ADDR_TIMER_RAN_OUT] = 0
+            self.pyboy.memory[ADDR_TIMER_PAUSED] = 0
+            self.pyboy.memory[ADDR_TIMER_ACTIVE] = 1
+            self.pyboy.memory[ADDR_D580] = 1
```

Comment:
Honestly i could not figure it out, i know its in the timer initialization logic so i just initialized it because its used somehow in the timer. The Pret repo just has it labeled as the address.